test/unit/snowflake_config_test.js

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+const assert = require('assert');
+const snowflake = require('./../../lib/snowflake');
+const ErrorCodes = require('./../../lib/errors').codes;
+const Logger = require('./../../lib/logger');
+const GlobalConfig = require('./../../lib/global_config');
+
+const LOG_LEVEL_TAGS = require('./../../lib/logger/core').LOG_LEVEL_TAGS;
+
+describe('Snowflake Configure Tests', function ()
+{
+  var originalConfig;
+
+  before(function ()
+  {
+    originalConfig = {
+      logLevel: Logger.getInstance().getLevelTag(),
+      insecureConnect: GlobalConfig.isInsecureConnect(),
+      ocspFailOpen: GlobalConfig.getOcspFailOpen(),
+      jsonColumnVariantParser: GlobalConfig.jsonColumnVariantParser,
+      xmlColumnVariantParser: GlobalConfig.xmlColumnVariantParser,
+    }
+  });
+
+  after(function ()
+  {
+    snowflake.configure(originalConfig);
+  });
+
+  ///////////////////////////////////////////////////////////////////////////
+  //// Test invalid arguments                                            ////
+  ///////////////////////////////////////////////////////////////////////////
+
+  var negativeTestCases =
+    [
+      {
+        name: 'invalid logLevel',
+        options: { logLevel: 'unsupported' },
+        errorCode: ErrorCodes.ERR_GLOBAL_CONFIGURE_INVALID_LOG_LEVEL
+      },
+      {
+        name: 'invalid insecureConnect',
+        options: { insecureConnect: 'unsupported' },
+        errorCode: ErrorCodes.ERR_GLOBAL_CONFIGURE_INVALID_INSECURE_CONNECT
+      },
+      {
+        name: 'invalid ocspMode',
+        options: { ocspFailOpen: 'unsupported' },
+        errorCode: ErrorCodes.ERR_GLOBAL_CONFIGURE_INVALID_OCSP_MODE
+      },
+      {
+        name: 'invalid json parser',
+        options: { jsonColumnVariantParser: 'unsupported' },
+        errorCode: ErrorCodes.ERR_GLOBAL_CONFIGURE_INVALID_JSON_PARSER
+      },
+      {
+        name: 'invalid xml parser',
+        options: { xmlColumnVariantParser: 'unsupported' },
+        errorCode: ErrorCodes.ERR_GLOBAL_CONFIGURE_INVALID_XML_PARSER
+      },
+    ];
+
+  var createNegativeITCallback = function (testCase)
+  {
+    return function ()
+    {
+      var error;
+
+      try
+      {
+        snowflake.configure(testCase.options);
+      }
+      catch (err)
+      {
+        error = err;
+      }
+      finally
+      {
+        assert.ok(error);
+        assert.strictEqual(error.code, testCase.errorCode);
+      }
+    };
+  };
+
+  var index, length, testCase;
+  for (index = 0, length = negativeTestCases.length; index < length; index++)
```

Comment:
we can simplify test cases to 
```
negativeTestCases.forEach(testCase => {it(testCase.name, createNegativeITCallback(testCase))})
```
or even inline createNegativeITCallback in test body

The same could be done with positive cases

Reply:
Nice, I'll use this for both the test cases. I put the callback function's body inside if that's what you meant